bech32-migration/pfmmiddleware/pfmmiddleware.go

```
@@ -0,0 +1,77 @@
+package pfmmiddleware
+
+import (
+	"encoding/json"
+	"fmt"
+
+	"github.com/cosmos/cosmos-sdk/codec"
+	storetypes "github.com/cosmos/cosmos-sdk/store/types"
+	sdk "github.com/cosmos/cosmos-sdk/types"
+	routertypes "github.com/cosmos/ibc-apps/middleware/packet-forward-middleware/v7/packetforward/types"
+	transfertypes "github.com/cosmos/ibc-go/v7/modules/apps/transfer/types"
+	"github.com/notional-labs/composable/v6/app/keepers"
+	"github.com/notional-labs/composable/v6/bech32-migration/utils"
+)
+
+func MigrateAddressBech32(ctx sdk.Context, storeKey storetypes.StoreKey, cdc codec.BinaryCodec, keepers *keepers.AppKeepers) {
+	ctx.Logger().Info("Migration of address bech32 for pfmmiddleware module begin")
+	totalAddr := uint64(0)
+
+	store := ctx.KVStore(storeKey)
+
+	allChannels := keepers.IBCKeeper.ChannelKeeper.GetAllChannels(ctx)
+	// if len(allChannels) == 0 {
+	//	allChannels = append(allChannels, types.IdentifiedChannel{
+	//		ChannelId: "channel-9",
+	//	})
+	//}
+	for _, channel := range allChannels {
+		channelKey := []byte(fmt.Sprintf("%s/", channel.ChannelId))
+		iterator := sdk.KVStorePrefixIterator(store, channelKey)
+		for ; iterator.Valid(); iterator.Next() {
+			totalAddr++
+			fullKey := iterator.Key()
+			if !store.Has(fullKey) {
+				continue
+			}
+			bz := store.Get(fullKey)
+			var inFlightPacket routertypes.InFlightPacket
+			cdc.MustUnmarshal(bz, &inFlightPacket)
+			inFlightPacket.OriginalSenderAddress = utils.SafeConvertAddress(inFlightPacket.OriginalSenderAddress)
+			var data transfertypes.FungibleTokenPacketData
+			if err := transfertypes.ModuleCdc.UnmarshalJSON(inFlightPacket.PacketData, &data); err != nil {
+				continue
```

i think instead of continuing, we can store the list of the failed packets during migration